pkcs11helpers/helpers.go

```diff
@@ -0,0 +1,226 @@
+package pkcs11helpers
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rsa"
+	"encoding/asn1"
+	"errors"
+	"fmt"
+	"math/big"
+
+	"github.com/miekg/pkcs11"
+)
+
+type PKCtx interface {
```

Comment:
The exported types in this file need documentation.

Seeing this factored out, it really does feel like it duplicates a lot of pkcs11key. Are you sure you wouldn't rather just use the library? It has the benefit that it's used somewhat more, so people are more likely to find bugs.

Reply:
I went back and forth about whether to use pkcs11key or not and landed here. I think there are a handful of reasons not to, the largest among them being that it makes testing a bit more painful and and requires either a number of changes to how the ceremony process works or changes to pkcs11key that would only be used in this specific context (which means we're back to it not getting touched by (many) other people in order to find bugs).

I think on the whole I eventually came down on the side of just writing the code here. It allows the gen-ca and gen-key packages to be somewhat more easy to audit, in terms of how they work, and have restricted interfaces in terms of what they can do and what signature formats etc that they support and what kinds of objects they can extract.